src/main/java/org/folio/dao/order/OrderPostgresDAO.java

```diff
@@ -23,4 +26,18 @@ public Future<Void> updateOrder(PurchaseOrder po, Conn conn) {
       .onFailure(t -> log.error("updateOrder failed for order with id {}", po.getId(), t))
       .mapEmpty();
   }
+
+  @Override
+  public Future<Long> getNextPoNumber(Conn conn) {
+    String sql = String.format("UPDATE %s SET last_number = last_number + 1 RETURNING last_number", ORDER_NUMBER_TABLE);
+    return conn.execute(sql)
+      .map(rowSet -> {
+        if (rowSet.rowCount() == 0) {
+          log.error("Could not get a new purchase order number (rowCount is 0)");
+          throw new HttpException(500, "Could not get a new purchase order number (rowCount is 0)");
```

Comment:
can we add query as parameter. It may be useful for troubleshooting

Reply:
I added it for the log, but the query will not change based on execution so I don't think it is worth it to make that exception creation more complex (especially since this would be an exceptional internal error).